source/MaterialXGenShader/Util.cpp

```diff
@@ -249,13 +250,21 @@ namespace
                     {
                         opaque = true;
                     }
-                    else if (transmission->getNodeName() == EMPTY_STRING && transmission->getInterfaceName() == EMPTY_STRING)
+                    else
                     {
-                        // Unconnected, check the value
-                        ValuePtr value = transmission->getValue();
-                        if (!value || (value->asA<float>() && isZero(value->asA<float>())))
+                        const string& tranmsInterfaceName = transmission->getInterfaceName();
+                        if (!tranmsInterfaceName.empty())
                         {
-                            opaque = true;
+                            transmissionInterfaceNames.insert(tranmsInterfaceName);
+                        }
+                        if (transmission->getNodeName() == EMPTY_STRING)
```

Comment:
Do not assume non-empty interface name means transparent. Just check the default value here and recheck later on at shaderref level.